crypto/Pos/Crypto/Signing/Redeem.hs

```diff
@@ -28,17 +30,18 @@ import           Pos.Crypto.Signing.Types.Redeem
 -- than the default IO generator.
 redeemKeyGen :: MonadRandom m => m (RedeemPublicKey, RedeemSecretKey)
 redeemKeyGen =
-    getRandomBytes 32 >>=
-    maybe err pure . redeemDeterministicKeyGen
-  where
-    err = error "Pos.Crypto.RedeemSigning.redeemKeyGen: createKeypairFromSeed_ failed"
+  do
+    sk <- Ed25519.generateSecretKey
+    return (RedeemPublicKey $ Ed25519.toPublic sk, RedeemSecretKey sk)
 
 -- | Create key pair deterministically from 32 bytes.
 redeemDeterministicKeyGen
     :: BS.ByteString
     -> Maybe (RedeemPublicKey, RedeemSecretKey)
 redeemDeterministicKeyGen seed =
-    bimap RedeemPublicKey RedeemSecretKey <$> Ed25519.createKeypairFromSeed_ seed
+  case Ed25519.secretKey $ (BA.convert seed :: BA.Bytes) of
+        CryptoPassed r -> Just (RedeemPublicKey $ Ed25519.toPublic r, RedeemSecretKey r)
+        CryptoFailed e -> error $ mappend "Pos.Crypto.Signing.Redeem.hs redeemDeterministicKeyGen failed because " (T.pack $ show e)
```

Comment:
This removes the safety of `Maybe` -- note that if we do:

```haskell
case redeemDeterministicKeygen someThing of
    Just a -> ...
    Nothing -> putStrLn "This will never happen!"
```

The `Nothing` branch will never execute, becuase the entire `case` expression will blow up with the error message.

This branch should either be `Nothing`, ignoring the error, or the type should be refactored to `Either CryptoError (RedeemPublicKey, RedeemSecretKey)`.